src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStoreCacheFileProvider.cs

```diff
@@ -56,9 +56,19 @@ public async Task SetCacheAsync(Stream stream, IFileStoreEntry fileStoreEntry, C
                 Directory.CreateDirectory(directory);
             }
 
-            using (var fileStream = File.Create(cachePath))
+            // A file download may fail, so a partially downloaded file should be deleted so the next request can reprocess.
+            try
+            {
+                using (var fileStream = File.Create(cachePath))
```

Comment:
Check if the file exists before creating it, then delete it. That should save some exceptions.